### PR TITLE
[build][graviton] Add buildspec parsing for Graviton buildspecs and dockerfiles

### PR DESCRIPTION
**Pull request by Qingzi-Lan**
*GitHub Issue #, if available:*
Note: If merging this PR should also close the associated Issue, please also add that Issue # to the Linked Issues section on the right. 

### Description
Add buildspec parsing for Graviton buildspecs and dockerfiles
## Label Checklist
- [x] I have added the project label for this PR (*<project_name>* or "Improvement")

## PR Checklist
- [x] I've prepended PR tag with frameworks/job this applies to : [mxnet, tensorflow, pytorch] | [ei/neuron] | [build] | [test] | [benchmark] | [ec2, ecs, eks, sagemaker]
- [x] If the PR changes affects SM test, I've modified dlc_developer_config.toml in my PR branch by setting sagemaker_tests = true and efa_tests = true
- [x] If this PR changes existing code, the change fully backward compatible with pre-existing code. (Non backward-compatible changes need special approval.)
- [ ] (If applicable) I've documented below the DLC image/dockerfile this relates to
- [ ] (If applicable) I've documented below the tests I've run on the DLC image
- [ ] (If applicable) I've reviewed the licenses of updated and new binaries and their dependencies to make sure all licenses are on the Apache Software Foundation Third Party License Policy Category A or Category B license list.  See [https://www.apache.org/legal/resolved.html](https://www.apache.org/legal/resolved.html).
- [ ] (If applicable) I've scanned the updated and new binaries to make sure they do not have vulnerabilities associated with them.

## Pytest Marker Checklist
- [ ] (If applicable) I have added the marker `@pytest.mark.model("<model-type>")` to the new tests which I have added, to specify the Deep Learning model that is used in the test (use `"N/A"` if the test doesn't use a model)
- [ ] (If applicable) I have added the marker `@pytest.mark.integration("<feature-being-tested>")` to the new tests which I have added, to specify the feature that will be tested
- [ ] (If applicable) I have added the marker `@pytest.mark.multinode(<integer-num-nodes>)` to the new tests which I have added, to specify the number of nodes used on a multi-node test
- [ ] (If applicable) I have added the marker `@pytest.mark.processor(<"cpu"/"gpu"/"eia"/"neuron">)` to the new tests which I have added, if a test is specifically applicable to only one processor type

#### EIA/NEURON Testing Checklist
* When creating a PR:
- [ ] I've modified `dlc_developer_config.toml` in my PR branch by setting `ei_mode = true` or `neuron_mode = true`

#### Benchmark Testing Checklist
* When creating a PR:
- [ ] I've modified `dlc_developer_config.toml` in my PR branch by setting `benchmark_mode = true`

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license. I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: src/main.py [hunk 1]
```diff
@@ -51,9 +55,16 @@ def main():
         and args.framework not in frameworks_to_skip
     )
 
+    # A GRAVITON dedicated builder will work if in GRAVITON mode and its framework has not been disabled
+    graviton_builder_enabled = (
```
Comment (SergTogul) [inline src/main.py:59]: nit: It's a good practice call boolean variables is_*

**Review by SergTogul — DISMISSED** [review]

File: tensorflow/training/docker/1.15/py3/cu110/Dockerfile.gpu [hunk 2]
```diff
@@ -34,7 +34,9 @@ ARG ESTIMATOR_URL=https://aws-tensorflow-binaries.s3-us-west-2.amazonaws.com/est
 # the nightly builds. Therefore, while updating the smdebug version, please ensure that the format is not disturbed.
 ARG SMDEBUG_VERSION=0.9.4
 
-RUN apt-get update && apt-get install -y --no-install-recommends --allow-unauthenticated \
+RUN apt-get update \
+ && apt-get -y upgrade --only-upgrade systemd \
+ && apt-get install -y --no-install-recommends --allow-unauthenticated \
```
Comment (saimidu) [inline tensorflow/training/docker/1.15/py3/cu110/Dockerfile.gpu:39]: @Qingzi-Lan Let's address this in a PR dedicated to it. This change is not related to the introduction of Graviton support.

**Review by saimidu — COMMENTED** [review]
LGTM. Please remove the changes to the TF 1.15 GPU dockerfile, and add the necessary tags - `[build][graviton]` - to the PR title, and then we can approve and merge this PR.

File: tensorflow/training/docker/1.15/py3/cu110/Dockerfile.gpu (hunk 2 above)
Comment (Qingzi-Lan) [reply tensorflow/training/docker/1.15/py3/cu110/Dockerfile.gpu:39]: right, will do.

**Comment by Qingzi-Lan** [comment]
> LGTM. Please remove the changes to the TF 1.15 GPU dockerfile, and add the necessary tags - `[build][graviton]` - to the PR title, and then we can approve and merge this PR.

DONE. and all failed tests are due to insufficient p3.xxx capacity.

**Review by arjkesh — APPROVED** [review]

**Review by saimidu — APPROVED** [review]
